src/libs/viewpoints/viewpoint_extractor.ts

```diff
@@ -0,0 +1,536 @@
+import { Service } from '@libs/service';
+import { ServiceIssue } from '@libs/service_issue';
+import { Viewpoint } from './viewpoint';
+
+/**
+ * Class that implements the pattern-based model clustering strategy proposed in:
+ *
+ * Figueiredo, G., Duchardt, A., Hedblom, M.M. and Guizzardi, G., 2018, May. Breaking into pieces:
+ * An ontological approach to conceptual model complexity management.
+ * In 2018 12th International Conference on Research Challenges in Information Science (RCIS) (pp. 1-10). IEEE.
+ * https://ieeexplore.ieee.org/iel7/8401323/8406639/08406642.pdf
+ *
+ * @author Guylerme Figueiredo
+ */
+
+
+export class ViewpointExtractor implements Service {
+  project: Project;
+
+  constructor(project: Project, _options?: any) {
+    this.project = project;
+
+    if (_options) {
+      console.log('Options ignored: this service does not support options');
+    }
+  }
+
+  run(): { result: any; issues?: ServiceIssue[] } {
+    const generatedDiagrams = this.buildAll();
+    this.project.addDiagrams(generatedDiagrams);
+
+    return {
+      result: this.project,
+      issues: null,
+    };
+  }
+
+  buildAll(): Diagram[] {
+    const allPatterns = [ClassStereotype.KIND, ClassStereotype.SUBKIND, ClassStereotype.ROLE, ClassStereotype.PHASE, ClassStereotype.RELATOR, ClassStereotype.MODE, 'NON_SORTAL'];
+    return allPatterns.map((pattern, index) => this.extractViewpoint(String(index), pattern))
+      .map((viewpoint) => viewpoint.createDiagram(this.project.model));
+  }
+
+  /**
+   * This methor will run the patter viewpoint extrator.
+   * 
+   * @param id The name of viewpoint extractor 
+   * @param pattern The viewpoint extractor approach based on OntoUML Patterns
+   * @returns A viewpoint based on a pattern choosen
+   */
+  extractViewpoint(id: string, pattern: string): Viewpoint {
+    let viewpoint = new Viewpoint('Viewpoint of ' + pattern);
+
+    switch (pattern) {
+      case ClassStereotype.KIND: viewpoint = this.kindView();
+        break;
+      case ClassStereotype.SUBKIND: viewpoint = this.subKindView();
+        break;
+      case ClassStereotype.ROLE: viewpoint = this.roleView();
+        break;
+      case ClassStereotype.PHASE: viewpoint = this.phaseView();
+        break;
+      case ClassStereotype.RELATOR: viewpoint = this.relatorView();
+        break;
+      case ClassStereotype.MODE: viewpoint = this.modeView();
+        break;
+      case 'NON_SORTAL': viewpoint = this.nonSortalView();
+        break;
+    }
+
+
+    return viewpoint;
+  }
+
+  /**
+   * This method filter all Kinds and related relations ando group them in a view
+   * 
+   * @returns A viewpoint extracted using the Kind Pattern
+   */
+  kindView(): Viewpoint {
+    const viewpoint = new Viewpoint('Kind View');
+
+    /**Step 1 Get all Kinds in model
+     * 
+     * Step 2 Get all relations between Kinds
+     * 
+     * Step 3 Get all generalizations between Kinds
+     */
+
```

Comment:
`Project` has a method `getClassesWithKindStereotype()`. Please check and reuse it.

Reply:
Done